### ISSUE TITLE
Paypal oauth is not working

### ISSUE THREAD

**Issue by anphu7492** [meanjs/mean#1055]
Hi there, 
I tried to add paypal authentication (with both sandbox and live options) but no luck. The application redirects to me the login page. Have anybody encountered this issue?

**Comment by 0xNacho** [comment]
could you provide more details about the issue? Did you create your PayPal app correctly?

**Comment by anphu7492** [comment]
Hi @0xNacho , I created the PayPal app from here: https://developer.paypal.com/developer/applications/
I also specified the return Url: http://localhost:3000/api/auth/paypal/callback
Is it the right way to do it? 
The app was created by my sandbox account (which is generated by paypal). The country of that sandbox account is US but my server is in EU. Does that matter?

**Comment by 0xNacho** [comment]
I don't think so.

After logging in, are you getting a URL like this ? 
``http://localhost:3000/?error_uri=http://localhost:3000&error_description=Client Validation Error: Invalid scope.&error=invalid_scope``

**Comment by anphu7492** [comment]
I'm not getting any errors. It doesn't even call the callback function:<code> function (req, accessToken, refreshToken, profile, done) {}</code> in paypal strategy.

**Comment by anphu7492** [comment]
Hi @0xNacho , 
Actually I got that error from the server logs. In the oauthCallback funtion, after passport authenticates the user == false, so the application redirects to the signin page. That why I didn't see the the url. Anyway, what could be the issue?

**Comment by 0xNacho** [comment]
Could you [gist](https://gist.github.com) your file?

**Comment by anphu7492** [comment]
https://gist.github.com/anphu7492/b34b28c881006db1585c#file-logs
Here it is

**Comment by anphu7492** [comment]
On line 4, I consoled value of <code>error</code> and <code>user</code> inside <code>passport.authenticate</code> function

**Comment by 0xNacho** [comment]
I'm sure your problem is in the callback URL.

Did you change your **paypal.callbackURL** param into your  ``/config/env/development.js`` file?

It should be: **/api/auth/paypal/callback**
 
Also, you need to specify your callbackURL into your PayPal App.Change your **Return URL** param as follows:
``http://localhost:3000/api/auth/paypal/callback```

Did this solve your problem?

**Comment by anphu7492** [comment]
Unfortunately not, my configuration was exactly like you said, in the config file /config/env/development.js file:
paypal: {
    clientID: process.env.PAYPAL_ID || 'myid',
    clientSecret: process.env.PAYPAL_SECRET || 'mysecret',
    callbackURL: '/api/auth/paypal/callback',
    sandbox: true
  }
callbackURL was specified like that as default. And I also specified the return URL on paypal developer dashboard. I also tried to clone the fresh meanjs project and configured paypal like my current project. I got the same issue
